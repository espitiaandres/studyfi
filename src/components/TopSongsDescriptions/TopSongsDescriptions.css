.tooltips {
    max-width: 400px;
}

.tooltip {
    background: #1ED760;
    border: none;
    border-radius: 1px;
    display: inline-block;
    font-size: 16px;
    margin: 0px 4px;  
    padding: 2px 10px;  
    text-decoration: none;
}

.seasonStyling {
    background: #A92921;
}

.seasonStylingAlt {
    background: #0A7533;
}
