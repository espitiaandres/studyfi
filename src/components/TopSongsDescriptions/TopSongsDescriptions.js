//
//  TopSongsDescriptions.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React from 'react';
import ReactTooltip from "react-tooltip";
import './TopSongsDescriptions.css';
import { 
    trackPopularityDesc,
    acousticnessDesc,
    danceabilityDesc,
    energyDesc,
    keySignatureDesc,
    instrumentalnessDesc,
    valenceDesc
} from '../../utils/TopTracksDefinitions';

const TopSongsDescriptions = ({ season }) => {
    let seasonStyling = season ? "seasonStyling" : "";
    let seasonStylingAlt = season ? "seasonStylingAlt" : "";

    return (
        <div className="columnDescriptions">
            <ReactTooltip className="tooltips" id="acousticness" type="light" effect="solid" place="top">
                <span>{acousticnessDesc}</span>
            </ReactTooltip>
            <button className={`tooltip ${seasonStyling}`} data-tip data-for="acousticness">
                acousticness
            </button>

            <ReactTooltip className="tooltips" id="popularity" type="light" effect="solid" place="top">
                <span>{trackPopularityDesc}</span>
            </ReactTooltip>
            <button className={`tooltip ${seasonStyling}`} data-tip data-for="popularity">
                popularity
            </button>

            <ReactTooltip className="tooltips" id="danceability" type="light" effect="solid" place="top">
                <span>{danceabilityDesc}</span>
            </ReactTooltip>
            <button className={`tooltip ${seasonStylingAlt}`} data-tip data-for="danceability">
                danceability
            </button>

            <ReactTooltip className="tooltips" id="energy" type="light" effect="solid" place="top">
                <span>{energyDesc}</span>
            </ReactTooltip>
            <button className={`tooltip ${seasonStyling}`} data-tip data-for="energy">
                energy
            </button>

            <ReactTooltip className="tooltips" id="keysignature" type="light" effect="solid" place="top">
                <span>{keySignatureDesc}</span>
            </ReactTooltip>
            <button className={`tooltip ${seasonStyling}`} data-tip data-for="keysignature">
                key signature
            </button>

            <ReactTooltip className="tooltips" id="instrumentalness" type="light" effect="solid" place="top">
                <span>{instrumentalnessDesc}</span>
            </ReactTooltip>
            <button className={`tooltip ${seasonStylingAlt}`} data-tip data-for="instrumentalness">
                instrumentalness
            </button>

            <ReactTooltip className="tooltips" id="valence" type="light" effect="solid" place="top">
                <span>{valenceDesc}</span>
            </ReactTooltip>
            <button className={`tooltip ${seasonStyling}`} data-tip data-for="valence">
                valence
            </button>
        </div>
    )
}

export default TopSongsDescriptions;
