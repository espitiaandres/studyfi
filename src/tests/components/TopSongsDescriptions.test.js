//
//  TopSongsDescriptions.test.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React from 'react';
import { shallow } from 'enzyme';
import TopSongsDescriptions from '../../components/TopSongsDescriptions/TopSongsDescriptions';

const topSongsDescriptionsPropsTest = {
    season: true
}

test('should render TopSongsDescriptions component correctly', () => {
    const wrapper = shallow(<TopSongsDescriptions />);
    expect(wrapper).toMatchSnapshot();
});

test('should render TopSongsDescriptions component correctly', () => {
    const wrapper = shallow(
        <TopSongsDescriptions 
            season={topSongsDescriptionsPropsTest.season}
        />
    );
    expect(wrapper).toMatchSnapshot();
});
