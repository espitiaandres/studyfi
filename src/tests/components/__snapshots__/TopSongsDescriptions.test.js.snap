// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TopSongsDescriptions component correctly 1`] = `ShallowWrapper {}`;

exports[`should render TopSongsDescriptions component correctly 2`] = `ShallowWrapper {}`;
